[WIP] Add user accessibility documentation (#15001)
jupyterlab/jupyterlab: docs/source/user/accessibility.rst

In addition to accessible third-party themes, JupyterLab helps users to create and apply custom themes. There are plenty of resources for developing a new theme including `this detailed blog post
<https://labs.quansight.org/blog/2020/12/jupyterlab-winter-theme>`_ and
the `Extension Development Guide
<https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html>`_

Text Scaling
^^^^^^^^^^^^

JupyterLab allows text scaling customization via the advanced settings for improving the legibility of certain parts of the interface like files. This improves the experience for users with visual impairment. For changing the font size options, please navigate via the main menu to
`Settings > Advanced Settings` and locate the text editor settings on the left panel to make the necessary adjustments.

It's important to note that changes made in this setting will exclusively impact either the Markdown viewer or the text editor, based on the selected configuration. Notably, adjustments here won't influence the font size in the notebook area or in the rest of the interface.

> However, users have the zoom browser functionality to increase the size of the interface elements of JupyterLab. This approach helps to improve readability in specific contexts. Take into account that the limit for font size is 100%.

Reviewer: Zoom browsers can go up to 400%, where did you found that the limit for font size is 100%?

-----------------------------------

These features are carefully crafted to provide seamless navigation, interaction, and customization options making JupyterLab a versatile platform that caters to diverse user needs.

-----------------------------------
Keyboard Navigation
-----------------------------------

Keyboard navigation lies at the core of JupyterLab's accessibility approach, allowing users to efficiently interact with the interface using keystrokes. It has naturally evolved to provide a quicker alternative to mouse-based interactions by the use of multiple accessibility tools throughout the interface that will be discussed in this section.

-----------------------------------
Keyboard Shortcuts
-----------------------------------

JupyterLab offers an extensive set of keyboard shortcuts that empower users to navigate through various functions and tools. From executing code cells to managing tabs, these shortcuts streamline workflows and enhance accessibility for users who rely on keyboard input.

For a comprehensive list of available shortcuts, users can explore the advanced settings by navigating to `Settings > Advanced Settings` in the JupyterLab main menu.

> .. figure:: ./desktop/quansight_labs/documentation/keyboard_shortcut_final.png

Reviewer: You need to update this route to link the images folder that will be under `docs/source/user/images/accessibility`

   :alt: an image showing how to add a shortcut in the advanced settings panel.

Within JupyterLab's Advanced Settings, default keyboard shortcuts are unmodifiable, though users can craft and modify their own shortcuts to align with their unique workflows. In order to create a new keyboard shortcut use the `add` button and enter the information requested,
Please note that the information can be removed and edited afterwards.

-----------------------------------
Command Palette
-----------------------------------

The Command Palette acts as a command center, offering a text-based interface for executing commands. It can be accessed from the `View` menu or using the keyboard shortcut `Ctrl + Shift + C`. This feature not only aids keyboard-centric users but also promotes efficiency and
accessibility, as users can swiftly access functionalities without navigating complex menus.

> .. figure:: ./desktop/quansight_labs/documentation/command_palette.png

Reviewer: You need to update this route to link the images folder that will be under `docs/source/user/images/accessibility`

Author: Done!

.. Copyright (c) Jupyter Development Team.
.. Distributed under the terms of the Modified BSD License.

> ==============================
> JupyterLab Accessibility Documentation
> ==============================

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
JupyterLab Accessibility Documentation
======================================
```

> ----------------
> Introduction
> ----------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Introduction
------------
```

> Welcome to the JupyterLab Accessibility Documentation, the guide for making the most of JupyterLab's inclusive features. As part of our commitment to providing an accessible and user-friendly environment, this documentation aims to empower users with insights, tips, and tools for enhancing accessibility within the JupyterLab interface.
> 
> In this guide, there is a wealth of information about key accessibility features, customization options, and best practices for navigating and interacting with JupyterLab. We are working towards ensuring that all users, regardless of their abilities, can seamlessly engage with
> JupyterLab's powerful capabilities.

Reviewer 3: These terms create too much expectation: as part of our commitment, user-friendly, empower, enhancing accessibility, regardless of their abilities, guide, wealth of information, seamlessly, powerful capabilities.

I would delete everything in this paragraph after "Welcome to the JupyterLab Accessibility Documentation" and go directly into the next paragraph, and delete the terms I identified above:

```suggestion
Welcome to the JupyterLab Accessibility Documentation. This page contains information about key accessibility features, customization options, and best practices for navigating and interacting with JupyterLab. This is part of ongoing efforts to improve the user experience for disabled users.
```

> -----------------------------------
> Key Accessibility Features
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Key Accessibility Features
--------------------------
```

> These features are carefully crafted to provide seamless navigation, interaction, and customization options making JupyterLab a versatile platform that caters to diverse user needs.

Reviewer 3: Marketing brochure language: carefully crafted, seamless, versatile, caters to diverse user needs.

Author: These features are implemented to help users navigate, interact and customize their experience in Jupyterlab. 

How about this? @gabalafou

Reviewer 3: This is better, yes. I'm a little unsure about framing this document in terms of features, though. I think that sends a message like "We built these things for **you**." Whereas I see this document as more like a place where various tips and tricks are gathered together in one place. It's definitely not an easy task to organize this information, so I applaud you on your efforts!

> -----------------------------------
> Keyboard Navigation
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Keyboard Navigation
^^^^^^^^^^^^^^^^^^^
```

> Keyboard navigation lies at the core of JupyterLab's accessibility approach, allowing users to efficiently interact with the interface using keystrokes. It has naturally evolved to provide a quicker alternative to mouse-based interactions by the use of multiple accessibility tools throughout the interface that will be discussed in this section.

Reviewer 3: How about replacing this paragraph with: Keyboard navigation is a core concern of JupyterLab's accessibility roadmap.

Author: This captures the essence of the section.

> -----------------------------------
> Keyboard Shortcuts
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Keyboard Shortcuts
^^^^^^^^^^^^^^^^^^
```

> JupyterLab offers an extensive set of keyboard shortcuts that empower users to navigate through various functions and tools. From executing code cells to managing tabs, these shortcuts streamline workflows and enhance accessibility for users who rely on keyboard input.

Reviewer 3: offers, extensive, empower, from x to y, streamline, enhance

Author: I don't think I get you on this. More context?

Reviewer 3: All of the inline comments from my code review 3 days ago were about marketing brochure language, so if you see a inline comment from me from 3 days ago with just a bunch of words, it means: these words or expressions are problematic and we need to remove them or tone them down.

> For a comprehensive list of available shortcuts, `users can check out this resource <https://jupyterlab.readthedocs.io/en/stable/user/commands.html#commands-list>`_, or explore the advanced settings by navigating to `Settings > Advanced Settings` in the JupyterLab main menu.

Reviewer 4: Is it also worth mentioning the shortcuts overview help from https://github.com/jupyterlab/jupyterlab/pull/14053?

.. figure:: ./images/accessibility/keyboard_shortcuts.png

>    :alt: The JupyterLab Advanced Settings Editor with the Keyboard Shortcuts section open. Under a selected keyboard shortcut listing there is a Add button to add another shortcut.

Reviewer 3: ```suggestion
   :alt: The JupyterLab Advanced Settings Editor with the Keyboard Shortcuts section open. Under a selected keyboard shortcut listing there is a button to add another shortcut.
```

> Within JupyterLab's Advanced Settings, default keyboard shortcuts are unmodifiable, though users can craft and modify their own shortcuts to align with their unique workflows. In order to create a new keyboard shortcut use the `add` button and enter the information requested,

Reviewer 3: "craft and"

> Within JupyterLab's Advanced Settings, default keyboard shortcuts are unmodifiable, though users can craft and modify their own shortcuts to align with their unique workflows. In order to create a new keyboard shortcut use the `add` button and enter the information requested,

Reviewer 4: Is this really the case? I think any shortcut can be modified?

Author: From my end, I wasn't able to modify default shortcuts. I was able to craft a new one and modify it.

Author: Is this correct?

Please note that the information can be removed and edited afterwards.

> -----------------------------------
> Command Palette
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Command Palette
^^^^^^^^^^^^^^^
```

> The `Command Palette <https://jupyterlab.readthedocs.io/en/latest/user/commands.html#command-palette>`_ acts as a list of all commands in JupyterLab, offering a text-based interface for executing commands. It can be accessed from the `View` menu or using the keyboard shortcut `Ctrl + Shift + C`. Using the shortcut `Ctrl + Shift + C` will change focus to the command palette no matter where users are in JupyterLab, so it may also be useful to move out of keyboard traps. This feature not only aids keyboard-centric users but also promotes efficiency and

Reviewer 3: not only x but also y, promotes, swiftly, without ... complex

accessibility, as users can swiftly access functionalities without navigating complex menus.

.. figure:: ./images/accessibility/Command_Palette.png
   :alt: JupyterLab notebook editor with the modal command palette opened. There is a list filter at the top and a truncated list of commands with shortcuts below.

> -----------------------------------
> Screen Reader Support
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Screen Reader Support
^^^^^^^^^^^^^^^^^^^^^
```

> While JupyterLab has taken strides to support screen reader compatibility, it's important to note that there hasn't been any extensive testing of this feature. Presently, screen readers such as `NVDA <https://www.nvaccess.org/download/>`_ and `JAWS <https://www.freedomscientific.com/products/software/jaws/>`_ on Windows, as well as `VoiceOver <https://support.apple.com/en-ng/guide/voiceover/welcome/mac>`_ on macOS, provide some level of compatibility. However, challenges exist, such as reading images without labels, equations, and links.

Reviewer 3: taken strides

> --------------------------------------------
> Tab Navigation and Tab Trapping
> --------------------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Tab Navigation and Tab Trapping
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The JupyterLab interface allows users to cycle through focusable elements using the `Arrow keys`, `Tab key` and `Shift key`. This sequential navigation allows the exploration of buttons, input fields, and interactive components in a given order.

In the main UI sections, including the notebook, console, sidebars and terminal, it is possible for users to become trapped within a specific area, limiting keyboard navigation. This is a work in progress and multiple efforts are being made to reduce the number of tab traps.

In case of encountering a Tab trap, users can use the Command Palette. By opening the Command Palette and searching for the desired command, this action enables the transition to other parts of JupyterLab's interface without relying solely on sequential tabbing.

> -----------------------------------
> Colour Themes
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Colour Themes
^^^^^^^^^^^^^
```

While JupyterLab provides default themes, it's important to note that these themes are not compliant with the `Web Content Accessibility Guidelines (WCAG) 2.0 <https://www.w3.org/TR/WCAG20/>`_. This means that some users might experience challenges in terms of colour contrast or readability.

> Nevertheless, `Jupyterlab Accessible Themes <https://github.com/Quansight-Labs/jupyterlab-accessible-themes>`_, a third-party extension provides an additional avenue for enhancing color accessibility in JupyterLab. This extension introduces two WCAG-compliant accessible themes to the JupyterLab ecosystem. Please note that this extension is compatible with JupyterLab 3.0.

Reviewer 3: enhancing

> In addition to accessible third-party themes, JupyterLab grants users the freedom to create and apply custom themes. There are plenty of resources for developing a new theme including `this detailed blog post

Reviewer 3: grants ... freedom

<https://labs.quansight.org/blog/2020/12/jupyterlab-winter-theme>`_ and
the `Extension Development Guide
<https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html>`_

> -----------------------------------
> Text Scaling
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Text Scaling
^^^^^^^^^^^^
```

> JupyterLab allows text scaling customization via the advanced settings for improving the legibility of certain parts of the interface like files. This enhances the experience for users with visual impairment. For changing the font size options, please navigate via the main menu to

Reviewer 3: enhances

`Settings > Advanced Settings` and locate the text editor settings on the left panel to make the necessary adjustments.

It's important to note that changes made in this setting will exclusively impact either the Markdown viewer or the text editor, based on the selected configuration. Notably, adjustments here won't influence the font size in the notebook area or in the rest of the interface.

> However, users have the zoom browser functionality to increase the size of not only the text size but also the interface elements of JupyterLab. This dynamic approach is a fast route to improve readability and increase workflow efficiency in specific contexts. Take into account that some parts of the interface may fail if the zoom is greater than 200%.

Reviewer 3: not only the x but also y, dynamic approach, fast route, workflow efficiency

-----------------------------------
Focus Indicators
-----------------------------------

Currently, JupyterLab does not provide focus indicators that fully adhere to Web Content Accessibility Guidelines (WCAG) standards. While keyboard navigation remains functional, the absence of WCAG-compliant focus indicators may impact the visual cues for users with accessibility needs.

> -----------------------------------
> Current Known Issues
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Current Known Issues
--------------------
```

As part of our accessibility commitment, we want to acknowledge and address the current known issues in JupyterLab's accessibility journey. While we are actively working to enhance the platform's accessibility features, some challenges may persist. Please refer to the following
links for seeing the current state and discussions taking place in the community.

-  `JupyterLab accessibility issues <https://github.com/jupyterlab/jupyterlab/issues?q=is%3Aopen+is%3Aissue+label%3Atag%3AAccessibility>`_

-  `Lumino accessibility issues <https://github.com/jupyterlab/lumino/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility>`_

-  `JupyterLab and Notebook accessibility audits <https://jupyter-accessibility.readthedocs.io/en/latest/audits/index.html>`_

> Please feel free to contribute to any of the listed projects, all the feedback will help to enhance JupyterLab's accessibility and ensure it accommodates diverse user needs.

Reviewer 3: enhance and ensure

> -----------------------------------
> Further Resources
> -----------------------------------

Reviewer 2: A [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Further Resources
-----------------
```

.. Copyright (c) Jupyter Development Team.
.. Distributed under the terms of the Modified BSD License.

==============================
JupyterLab Accessibility Documentation
==============================

----------------
Introduction
----------------

Welcome to the JupyterLab Accessibility Documentation, the guide for making the most of JupyterLab's inclusive features. As part of our commitment to providing an accessible and user-friendly environment, this documentation aims to empower users with insights, tips, and tools for enhancing accessibility within the JupyterLab interface.

In this guide, there is a wealth of information about key accessibility features, customization options, and best practices for navigating and interacting with JupyterLab. We are working towards ensuring that all users, regardless of their abilities, can seamlessly engage with
JupyterLab's powerful capabilities.

-----------------------------------
Key Accessibility Features
-----------------------------------

> These features are carefully crafted to provide seamless navigation, interaction, and customization options making JupyterLab a versatile platform that caters to diverse user needs.

Reviewer 2: I think that makes sense and provides more concrete examples of what will be covered. That works for me!

-----------------------------------
Keyboard Navigation

> -----------------------------------

Author: Thank you, Isabella, for your feedback. Your observation is correct. The hierarchy you've outlined aligns with our intended structure for the documentation. We appreciate your willingness to help format the headings to maintain proper hierarchy. 

Please feel free to provide your suggestions, and we'll ensure the headings are organized accordingly.

Keyboard navigation lies at the core of JupyterLab's accessibility approach, allowing users to efficiently interact with the interface using keystrokes. It has naturally evolved to provide a quicker alternative to mouse-based interactions by the use of multiple accessibility tools throughout the interface that will be discussed in this section.

> 

Reviewer 2: I'd defer to @gabalafou's feedback here then; I think he is both closer to current keyboard efforts and more aware of what might be useful information here.

I'm in favor of the idea of noting this is a progress, though. Something like 
> There are currently no specialized keyboard navigation settings in JupyterLab. Navigate it like you would any other webpage in your browser. 

could work if that seems accurate.

> 

Reviewer 3: I think JupyterLab has done some things to improve keyboard navigation, even specifically as it relates to accessibility. For example, if you're in the JupyterLab UI and you open the VoiceOver rotor, you see the following:

<img width="1728" alt="VoiceOver rotor in JupyterLab when the file browser and a notebook are open, lists the following landmarks: banner, main navigation, main sidebar complementary, File Browser Section region, file browser navigation, side panel content region, main, notebook actions navigation, notebook content region, alternate sidebar complementary, content information" src="https://github.com/jupyterlab/jupyterlab/assets/317883/20c375ae-8db3-4e59-ac4b-a8f7fa1eb212">

I think there's certainly room for improvement—for example, "main navigation" should probably be "top menu bar"—but these landmarks wouldn't exist, as far as I understand, if people had gone through and made specific accessibility-related changes to the HTML markup for the UI.

And as far as I know, every screen reader has the ability to show these landmarks and make it easier for screen reader users to jump from one part of the UI to another. Is this what we mean by keyboard navigation?

> 

Reviewer: I think we can mention this in the screen reader section

-----------------------------------
Keyboard Shortcuts
-----------------------------------

JupyterLab offers an extensive set of keyboard shortcuts that empower users to navigate through various functions and tools. From executing code cells to managing tabs, these shortcuts streamline workflows and enhance accessibility for users who rely on keyboard input.

> For a comprehensive list of available shortcuts, users can explore the advanced settings by navigating to `Settings > Advanced Settings` in the JupyterLab main menu.

Reviewer 4: We also have a list online: https://jupyterlab.readthedocs.io/en/stable/user/commands.html#commands-list, worth mention and linking?

.. figure:: ./desktop/quansight_labs/documentation/keyboard_shortcut_final.png
   :alt: an image showing how to add a shortcut in the advanced settings panel.

Within JupyterLab's Advanced Settings, default keyboard shortcuts are unmodifiable, though users can craft and modify their own shortcuts to align with their unique workflows. In order to create a new keyboard shortcut use the `add` button and enter the information requested,
Please note that the information can be removed and edited afterwards.

-----------------------------------
Command Palette
-----------------------------------

The Command Palette acts as a command center, offering a text-based interface for executing commands. It can be accessed from the `View` menu or using the keyboard shortcut `Ctrl + Shift + C`. This feature not only aids keyboard-centric users but also promotes efficiency and
accessibility, as users can swiftly access functionalities without navigating complex menus.

> 

Author: Oh! yeah, just figured. Thank you.

.. figure:: ./desktop/quansight_labs/documentation/command_palette.png
   :alt: an image showing the activated command palette modal.

-----------------------------------
Screen Reader Support
-----------------------------------

> While JupyterLab has taken strides to support screen reader compatibility, it's important to note that there hasn't been any extensive testing of this feature. Presently, screen readers such as `NVDA <https://www.nvaccess.org/download/>`_ and `JAWS <https://www.freedomscientific.com/products/software/jaws/>`_ on Windows, as well as `VoiceOver <https://support.apple.com/en-ng/guide/voiceover/vo2682/mac#:~:text=You%20can%20also%20turn%20VoiceOver,then%20press%20the%20Space%20bar.>`_ on macOS, provide some level of compatibility. However, challenges exist, such as reading images without labels, equations, and links.

Reviewer 4: ```suggestion
While JupyterLab has taken strides to support screen reader compatibility, it's important to note that there hasn't been any extensive testing of this feature. Presently, screen readers such as `NVDA <https://www.nvaccess.org/download/>`_ and `JAWS <https://www.freedomscientific.com/products/software/jaws/>`_ on Windows, as well as `VoiceOver <https://support.apple.com/en-ng/guide/voiceover/welcome/mac>`_ on macOS, provide some level of compatibility. However, challenges exist, such as reading images without labels, equations, and links.
```

Suggestion to avoid linking to any specific version, and including the highlight text

--------------------------------------------
Tab Navigation and Tab Trapping
--------------------------------------------

> The JupyterLab interface allows users to cycle through focusable elements using the `Arrow keys`, `Tab key` and `Shift key`. This sequential navigation allows the exploration of buttons, input fields, and interactive components in a given order.

Reviewer 4: There is a dedicated [`:kbd:` role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-kbd) in sphinx for keyboard shortcuts.

-----------------------------------
Colour Themes
-----------------------------------

While JupyterLab provides default themes, it's important to note that these themes are not compliant with the `Web Content Accessibility Guidelines (WCAG) 2.0 <https://www.w3.org/TR/WCAG20/>`_. This means that some users might experience challenges in terms of colour contrast or readability.

Nevertheless, `Jupyterlab Accessible Themes <https://github.com/Quansight-Labs/jupyterlab-accessible-themes>`_, a third-party extension provides an additional avenue for enhancing color accessibility in JupyterLab. This extension introduces two WCAG-compliant accessible themes to the JupyterLab ecosystem. Please note that this extension is compatible with JupyterLab 3.0.

In addition to accessible third-party themes, JupyterLab grants users the freedom to create and apply custom themes. There are plenty of resources for developing a new theme including `this detailed blog post
<https://labs.quansight.org/blog/2020/12/jupyterlab-winter-theme>`_ and
the `Extension Development Guide

> <https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html>`_

Reviewer 4: I guess that https://github.com/jupyterlab-contrib/jupyterlab-theme-editor is worth mention too.

-----------------------------------
Text Scaling
-----------------------------------

JupyterLab allows text scaling customization via the advanced settings for improving the legibility of certain parts of the interface like files. This enhances the experience for users with visual impairment. For changing the font size options, please navigate via the main menu to
`Settings > Advanced Settings` and locate the text editor settings on the left panel to make the necessary adjustments.

It's important to note that changes made in this setting will exclusively impact either the Markdown viewer or the text editor, based on the selected configuration. Notably, adjustments here won't influence the font size in the notebook area or in the rest of the interface.

> However, users have the zoom browser functionality to increase the size of not only the text size but also the interface elements of JupyterLab. This dynamic approach is a fast route to improve readability and increase workflow efficiency in specific contexts. Take into account that some parts of the interface may fail if the zoom is greater than 200%.

Author: From a second look, the limit for text scaling is 100%. Seems 200% was a typo

And this is prevalent in text files, markdown files and Python files.

> However, users have the zoom browser functionality to increase the size of not only the text size but also the interface elements of JupyterLab. This dynamic approach is a fast route to improve readability and increase workflow efficiency in specific contexts. Take into account that some parts of the interface may fail if the zoom is greater than 200%.

Author: Make the correction

> -----------------------------------
> Focus Indicators
> -----------------------------------
> 
> Currently, JupyterLab does not provide focus indicators that fully adhere to Web Content Accessibility Guidelines (WCAG) standards. While keyboard navigation remains functional, the absence of WCAG-compliant focus indicators may impact the visual cues for users with accessibility needs.

Reviewer 2: If we're not removing the section, here's a [heading restructuring suggestion](https://github.com/jupyterlab/jupyterlab/pull/15001/files#r1303598452)

```suggestion
Focus Indicators
^^^^^^^^^^^^^^^^

Currently, JupyterLab does not provide focus indicators that fully adhere to Web Content Accessibility Guidelines (WCAG) standards. While keyboard navigation remains functional, the absence of WCAG-compliant focus indicators may impact the visual cues for users with accessibility needs.
```
